corehq/apps/repeaters/models.py

```diff
@@ -324,6 +261,13 @@ class FormRepeater(Repeater):
 
     """
 
+    class Formats(object):
+        formats = {
+            'form_xml': (FormRepeaterXMLPayloadGenerator, _('XML')),
+            'form_json': (FormRepeaterJsonPayloadGenerator, _('JSON')),
+        }
+        default_format = 'form_xml'
```

Comment:
@proteusvacuum Yeah, so I thought of some version of that too earlier. Not sure what you mean by

> I guess that doesn't work with class instantiation.

exactly, but the thing that broke it for me is that this `.formats` property we're trying to create is primarily a _class_ property and not an instance property. `@property` only works to create _instance_ properties; `Repeater.formats` in the example above would be just the property object that `property` returns.

We could totally just make it `_`-prefixed and be fine referencing it outside the class or through a `get_formats()` classmethod.

In addition but separately, it'd kind of make more sense to call the method `generators` (or `_generators`) rather than `formats`. Not sure what you think.